### DISCUSSION TITLE
Mac Permissions pop up for every new .cpp file

### DISCUSSION THREAD

**Discussion by kornnchenn** [microsoft/vscode-cpptools#11904] [Q&A]
So I figure this is an issue with how I setup something.. but every time I compile a new .cpp file in VS Code (on MacOS Ventura 13.6.3) I get a pop up asking if I would like to grant permissions to my Documents folder. 

In Privacy it looks like it's creating a new app for every .cpp file? Shown are helloworld, hw2part1, and particles. This is only kind of annoying but I should probably fix it since eventually this is gonna be filled with all files I've made .. I assume that it should be running as Terminal, or as VS Code, which already have access to my Documents folder?

<img width="499" alt="Screenshot 2024-01-28 at 3 35 10 PM" src="https://github.com/microsoft/vscode-cpptools/assets/77260112/f26eec7f-6f3b-42d4-906b-de81a537fc5b">

Thanks for your help!

P.S. VS Code is called "Visual Studio Code 2" in my computer cause I had previously had it downloaded, deleted it, and then reinstalled and for some reason it's called that now... "Visual Studio Code" does not appear in my Applications folder or Launchpad (the name with the 2 does) but they are both there as an option when you right click > file > open with .. and they both open the same thing... idk that's a separate issue I assume lol

**Comment by Dark-Schneider-666** [comment]
Hello,
Compiling over a OS protected folder (Documents, external drives, etc.) has that behavior. Some solutions are:
- Use a non-protected folder, like your home, to place your project.
- Specify the output target in a non-protected folder, using the tasks.json, something like:
```
"args": [
				"-std=c++17",
				"-stdlib=libc++",
				"-fcolor-diagnostics",
				"-fansi-escape-codes",
				"-g",
				"${file}",
				"-o",
				//"${fileDirname}/${fileBasenameNoExtension}"
				"/tmp/${fileBasenameNoExtension}"
			],
```
Notice the change for the parameter "-o". Compiling with that you will be no asked for permission.
